tensorboard/examples/plugins/example_raw_scalars/tensorboard_plugin_example_raw_scalars/util.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utilities for the plugin backend."""
+
+import os
+
+
+def can_serve_from_static(path):
+    """Returns whether a filepath can be served from the static folder.
+
+    Args:
+        path: A string file path.
+    """
+    abs_path = os.path.abspath(path)
+    container_dir_prefix = os.path.abspath("./static") + os.sep
```

Comment:
As written, doesn’t this still let an attacker glean information about
the host directory setup? For example, the following test fails:

```python
# Should be false, but is true in the Bazel test runner!
self.assertFalse(
    can_serve_from_static(
        "static/../../org_tensorflow_tensorboard/static/index.js"
    )
)                                                                     
```

…because in the Bazel test environment the working directory is
`org_tensorflow_tensorboard`. You can use further `../EXPECTED_DIR`
components to determine the full path. This shouldn’t be possible.

Logically, we should be able to determine whether a path is safe without
making any syscalls, and doing so will likely be more clearly correct.
(Does your existing code do the right thing on Windows? You’ve used both
`os.path.join` and hard-coded forward slashes, plus HTTP paths always
use `/` as a path component separator, regardless of platform.)

Instead of using `os.path.abspath`, consider using `posixpath.normpath`,
which is Python’s equivalent of `require("path").posix.normalize` as
opposed to `require("path").resolve`.

Reply:
Thanks for the suggestion, it's now using `posixpath.normpath`!

> Does your existing code do the right thing on Windows?

I still need to check this on a Windows machine.  My thinking is that we could use `os.path.join(os.path.dirname(__file__), os.path.normpath(...))`, which should always join the result of 2 OS-specific URLs.